integrations/jira/jira/client.py

```diff
@@ -44,24 +55,98 @@ def __init__(self, jira_url: str, jira_email: str, jira_token: str) -> None:
 
     @staticmethod
     def _generate_base_req_params(
-        maxResults: int = 0, startAt: int = 0
+        maxResults: int = 50, startAt: int = 0
     ) -> dict[str, Any]:
         return {
             "maxResults": maxResults,
             "startAt": startAt,
         }
 
-    async def _get_paginated_projects(self, params: dict[str, Any]) -> dict[str, Any]:
-        project_response = await self.client.get(
-            f"{self.api_url}/project/search", params=params
-        )
-        project_response.raise_for_status()
-        return project_response.json()
+    async def _make_paginated_request(
+        self,
+        url: str,
+        params: dict[str, Any] = {},
+        data_key: str = "values",
+        is_last_function: typing.Callable[
+            [dict[str, Any]], bool
+        ] = lambda response: response["isLast"],
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        params = {**self._generate_base_req_params(), **params}
+        is_last = False
+        logger.info(f"Making paginated request to {url} with params: {params}")
+        while not is_last:
+            try:
+                response = await self.client.get(url, params=params)
+                response.raise_for_status()
+                response_data = response.json()
+                values = response_data[data_key]
```

Comment:
i dont like that data_key thingy
please just return the response and let the handling function to get the data it wants